README-dev.md

```diff
@@ -0,0 +1,266 @@
+* Make changes to the Alive-lsp code (like logging `"Happy!"` at startup).
+* Use the **Developer: Reload Window** command to restart the extension.
+* Test the Alive functionality supported by the server.
+* Log data will be in the `Output` view.
+
+#### Testing Alive *and* Alive-lsp
+
+It is often necessary to work on both Alive and Alive-lsp at the same time.
+This configuration allows changes to be made to both codebases.
+
+In the Alive-lsp directory:
+* Set the `alive.lsp.install.path` to the root of the Alive-lsp code.
+
+In the Alive directory:
+* Bring up VSCode.
+* Make changes to the Alive code.
+* Start the debugger (e.g. by pressing the `F5` key)
+  to generate a VSCode Extension Development Host window.
+* If necessary tell VSCode to open the Alive-lsp project in the Extension Development Host window.
+* Make changes to the Alive-lsp code.
+* Use the **Developer: Reload Window** command to restart the extension.
+* Exercise the Alive functionality supported by the server.
+* Log data from Alive-lsp will be in the `Output` view.
+
+#### Testing via REPL
+
+This method may help in cases where the REPL is crashing inexplicably.
+Consider this the configuration of last resort.
+
+In the Alive-lsp directory:
+* Start the server in an SBCL REPL using the same port number:
+```
+        sbcl --eval "(asdf:load-system :alive-lsp)" \
+             --eval "(alive/server:start :port 8006)"
+```
+
+In the Alive directory:
+* Set the `alive.lsp.remote.host` to `"127.0.0.1"`.
+* Set the `alive.lsp.remote.port` to some number (e.g. `8006`).
+* Use the **Developer: Reload Window** command to restart the extension.
+* Exercise the server via the Alive extension to invoke the broken server code.
+  Some of this (e.g. acquiring defined packages)
+  may happen automatically at startup.
+
+In the Alive-lsp directory:
+* The REPL will crash and complain
+  `The current thread is not at the foreground`.
+* Execute `(sb-thread:release-foreground)`.
+  The REPL prompt may not be visible, just enter the command.
+* Use the Common Lisp
+  [interactive debugger](https://lispcookbook.github.io/cl-cookbook/debugging.html#the-interactive-debugger).
+
+The REPL can be run in a shell window from the Alive-lsp directory or in the
+`Terminal` view at the bottom of the VSCode workbench.
+Executing in a shell window may be marginally less confusing.
+
+With this configuration it is possible to add `(break)` calls in the code to cause a break in the REPL.
+Note that breaks in some areas will prevent the Alive extension from completing startup.
+For example, during startup Alive will request all known packages, ASDF systems, and threads.
+Breaking within the function `handle-list-pkgs` will halt processing of the packages request and
+the Alive extension startup will be delayed and (maybe) eventually fail.
+The **Locking Up the Alive Extension** section below addresses recovering from this sort of problem
+without resorting to `vim`.
```

Comment:
Resorting to vim is never necessary. If the extension gets in that bad of a state, disable it so that VSCode treats lisp files as plain text, make whatever changes get it back to working, and turn the extension back on.

Reply:
Removed the cheeky reference to `vim`.